src/ucs/sys/sock.c

```diff
@@ -368,26 +370,28 @@ ucs_status_t ucs_socket_server_init(const struct sockaddr *saddr, socklen_t sock
         goto err_close_socket;
     }
 
-    if (ports_range_start && (ports_range_start == ports_range_end)) {
+    if ((ports_range_start != 0) && (ports_range_start == ports_range_end)) {
         /* non zero single value port */
         retry = 0;
     } else {
         /* port value is zero or a range of ports is given */
         retry = 1;
     }
 
+    rand_seed = ucs_generate_uuid((uintptr_t)saddr);
+
     do {
-        if (ports_range_start || ports_range_end) {
-            status = ucs_sockaddr_set_port((struct sockaddr *)saddr,
-                                           /* coverity[dont_call] */
-                                           (uint16_t)((rand() %
-                                                      (ports_range_end -
-                                                       ports_range_start + 1)) +
-                                                      ports_range_start));
+        if ((ports_range_start != 0) || (ports_range_end != 0)) {
+            /* coverity[dont_call] */
```

Comment:
is the coverity suppression still needed here?
I see other places don't have it
can we drop it?

Reply:
right, i'll remove this coverity suppression. thanks!